tests/test_transport.py

```diff
@@ -927,3 +929,49 @@ def test_channel_send_memoryview(self):
             # sendall() accepts a memoryview instance
             chan.sendall(memoryview(data))
             self.assertEqual(sfile.read(len(data)), data)
+
+    def test_global_request_timeout(self):
+        """
+        verify global_request timeouts.
+        """
+        self.setup_test_server()
+
+        with mock.patch.object(self.server, 'check_port_forward_request',
+                               new=lambda x, y: time.sleep(2)):
```

Comment:
Should not use `time.sleep()` in tests. With the real timeout of 1 second below, that ensures this test will always be slow. Can either mock `time.time()` in the function for the same effect, or pass in a timer at some point.